docs/install/install-from-source.rst

```diff
@@ -0,0 +1,214 @@
+Install from source (developer install)
+=========================================
+
+.. contents::
+
+This page describes the build and installation of PyBaMM from the source code, available on GitHub. Note that this is **not the recommended approach for most users** and should be reserved to people wanting to participate in the development of PyBaMM, or people who really need to use bleeding-edge feature(s) not yet available in the latest released version. If you do not fall in the two previous categories, you would be better off installing PyBaMM using pip or conda.
+
+Lastly, familiarity with the python ecosystem is recommended (pip, virtualenvs).
+Here is a gentle introduction/refresher: `Python Virtual Environments: A Primer <https://realpython.com/python-virtual-environments-a-primer/>`_.
+
+
+Prerequisites
+---------------
+
+The following instructions are valid for both GNU/Linux distributions and MacOS.
+If you are running Windows, consider using the `Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`_.
+
+To obtain the PyBaMM source code, clone the GitHub repository
+
+.. code:: bash
+
+	  git clone https://github.com/pybamm-team/PyBaMM.git
+
+or download the source archive on the repository's homepage.
+
+To install PyBaMM, you will need:
+
+- Python 3 (PyBaMM supports versions 3.6, 3.7 and 3.8)
+- The python headers file for your current python version.
+- A BLAS library (for instance `openblas <https://www.openblas.net/>`_).
+- A C compiler (ex: ``gcc``).
+- A Fortran compiler (ex: ``gfortran``).
+
+On Ubuntu, you can install the above with
+
+.. code:: bash
+
+	  sudo apt install python3.X python3.X-dev libopenblas-dev gcc gfortran
+
+Where ``X`` is the version sub-number.
+
+On MacOS,
+
+.. code:: bash
+
+	  brew install python openblas gcc gfortran
+
+Finally, we recommend using `Tox <https://tox.readthedocs.io/en/latest/>`_.
+You can install it with
+
+.. code:: bash
+
+	  python3.X -m pip install --user tox
+
+Depending on your operating system, you may or may not have ``pip`` installed along python.
+If ``pip`` is not found, you probably want to install the ``python3-pip`` package.
+
+Installing the build-time requirements
+--------------------------------------
+
+Using Tox (recommended)
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+	  # in the PyBaMM/ directory
+	  tox -e build-requires
+
+This will download, compile and install the SuiteSparse and SUNDIALS libraries.
+Both libraries are installed in ``~/.local``.
+
+Using Homebrew (recommended for MacOS users)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
not clear if this should be done instead of tox or as an alternative on MacOS

Reply:
I specified the titled of the section above.